server/src/data_sources/offline_provider.rs

```diff
@@ -30,6 +30,15 @@ impl TokenSource for OfflineProvider {
         Ok(self.valid_tokens.values().cloned().collect())
     }
 
+    async fn get_valid_tokens(&self) -> EdgeResult<Vec<EdgeToken>> {
```

Comment:
Idle thoughts but it looks like we might end up writing this same method over and over. Might be worth implementing this as a trait on `Vec<EdgeToken>`. Not something for this PR